pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1362,7 +1362,7 @@ func (l *LibvirtDomainManager) getDomainSpec(dom cli.VirDomain) (*api.DomainSpec
 	if err != nil {
 		return nil, err
 	}
-	return util.GetDomainSpec(state, dom)
+	return util.GetDomainSpecWithRuntimeInfo(state, dom)
```

Comment:
what invocation of the `l.getDomainSpec()` is causing issues?

My concern is that this is changing a helper function that's used all throughout manager.go. It's unclear to me if there are any side effects from this change we may not be aware of.

Reply:
Indeed. We have to be very careful here. Libvirt does not always provide all xmls. It is easy to hit corner-cases, which we already did in the past, and where we sometimes even had crashes in virt-launcher then. We need at least a fallback call if the xml with the runtime info is not present.

Reply:
I've changed this code because of that reason... I've added a dedicated function to the command server instead of modifying the global one. I think that it is safer and this function is only used during timed resyncs of the domain.

Reply:
@davidvossel this was the problematic place: https://github.com/kubevirt/kubevirt/pull/4395/files#diff-1f215452754d7113cecfc01d2b0eeba0ba40cf03d20ad8816f4cc5fc0991b7b7R398

Reply:
oh interesting, that makes sense

Reply:
@rmohr do you remember what situation would cause virt-launcher to crash when retrieving xml when runtime info isn't present?

Based on your experience, do you think it makes sense for us to have two separate functions here, or should we have one function that always attempts to get runtime info, and falls back to getting non runtime info if an error occurs?